azurerm/internal/services/synapse/synapse_spark_pool_resource.go

```diff
@@ -206,55 +206,55 @@ func resourceArmSynapseBigDataPoolCreateUpdate(d *schema.ResourceData, meta inte
 
 	future, err := client.CreateOrUpdate(ctx, workspaceId.ResourceGroup, workspaceId.Name, name, bigDataPoolInfo, nil)
 	if err != nil {
-		return fmt.Errorf("creating Synapse BigDataPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+		return fmt.Errorf("creating Synapse Spark Pool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
 	}
 
 	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
-		return fmt.Errorf("waiting on creating future for Synapse BigDataPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+		return fmt.Errorf("waiting on creating future for Synapse Spark Pool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
 	}
 
 	resp, err := client.Get(ctx, workspaceId.ResourceGroup, workspaceId.Name, name)
 	if err != nil {
-		return fmt.Errorf("retrieving Synapse BigDataPool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
+		return fmt.Errorf("retrieving Synapse Spark Pool %q (Resource Group %q / workspaceName %q): %+v", name, workspaceId.ResourceGroup, workspaceId.Name, err)
 	}
 
 	if resp.ID == nil || *resp.ID == "" {
-		return fmt.Errorf("empty or nil ID returned for Synapse BigDataPool %q (Resource Group %q / workspaceName %q) ID", name, workspaceId.ResourceGroup, workspaceId.Name)
+		return fmt.Errorf("empty or nil ID returned for Synapse Spark Pool %q (Resource Group %q / workspaceName %q) ID", name, workspaceId.ResourceGroup, workspaceId.Name)
 	}
 
 	d.SetId(*resp.ID)
-	return resourceArmSynapseBigDataPoolRead(d, meta)
+	return resourceArmSynapseSparkPoolRead(d, meta)
 }
 
-func resourceArmSynapseBigDataPoolRead(d *schema.ResourceData, meta interface{}) error {
+func resourceArmSynapseSparkPoolRead(d *schema.ResourceData, meta interface{}) error {
 	client := meta.(*clients.Client).Synapse.BigDataPoolClient
```

Comment:
Can we change the client name to match the new resource name here?

```suggestion
	client := meta.(*clients.Client).Synapse.SparkPoolClient
```